File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0; 
        int right = s.length() - 1;  // Fixed: should be length - 1
        
        while (left < right) {
            // Skip non-alphanumeric characters from left
            if (!isalnum(s[left])) {
                left++;
                continue;
            }
            
            // Skip non-alphanumeric characters from right
            if (!isalnum(s[right])) {
                right--;
                continue;
            }
            
            // Compare alphanumeric characters
            if (tolower(s[left]) != tolower(s[right])) {
                return false;
            }
            
            // Move both pointers
            left++;
            right--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        pointerS, pointerT = 0, 0

        while pointerS < len(s) and pointerT < len(t):
            if s[pointerS] == t[pointerT]:
                pointerT += 1
            pointerS += 1

        return len(t) - pointerT